### PR TITLE
Improve assembly

### PR DESCRIPTION
**Pull request by SilasK**
first merge #60

@brwnj  I will merge when finished, but I'd like to have your input. 

- I split the function `align_reads_to_final_contigs` into two, to optimize the workflow.
- I clarified some of the parameters in `featureCounts`, e.g. what happen when one reads overlaps two genes. 

 When using the default treshold ( `CONTIG_MIN_ID=0.76`) all potential mapping positions above threshold are counted. Resulting in counting one read multiple times. The gene count have now a correction for this thanks to the `--fraction` param but the contig abundance are not corrected. And they are used for binning. 

Therefore I think that `CONTIG_COUNT_MULTI_MAPPED_READS=False` is better than `CONTIG_COUNT_MULTI_MAPPED_READS=True`. 

using `CONTIG_COUNT_MULTI_MAPPED_READS` one can define how ambiguous reads are handled in the two rules `align_reads_to_final_contigs` and `featureCounts`:
`CONTIG_COUNT_MULTI_MAPPED_READS=False` results in mapping only the best to the contigs and countig them. 

Alternative ways are possible, what do you think?

### CODE REVIEW

**Comment by SilasK** [comment]
When I did assembly I got a lot of contigs and thought that it would be intelligent to use a scaffolder. My colleges told me that [SSPACE](https://github.com/nsoranzo/sspace_basic) should be the best. @brwnj what do you think?

**Comment by SilasK** [comment]
@brwnj have you seen that one can calculate the abundance of genes in one shot with `pileup.sh`but I don't think it has all the options like `featureCounts`.

**Comment by brwnj** [comment]
I think that's a good separation of functionality in terms of counts to contigs for their qc and then counts to genes for quantification.

I've never used SSPACE for scaffolding, but agree that it would help with often fragmented assemblies. I would perform scaffolding on all contigs, prior to performing any filtering, then perform downstream processes on the scaffolds.

In my experience, contigs ends should be trimmed at least 100 bp to remove assembly artifacts, so I would recommend a step prior to scaffolding that performs:

CONTIG_TRIM_BP: 100

This currently occurs in the filter_by_coverage rule.

File: atlas/rules/assemble.snakefile [hunk 1]
```diff
@@ -1012,32 +1003,53 @@ rule align_reads_to_filtered_contigs:
...
+        """
+            pileup.sh ref={input.fasta} in={input.sam} threads={threads} \
             -Xmx{resources.mem}G covstats={output.covstats} \
-            hist={output.covhist} basecov={output.basecov} physcov 2>> {log}
-
-            #samtools view -u -f4 {output.sam} | samtools bam2fq -s unmapped.se.fq - > unmapped.pe.fq
-
-               """
-
+            hist={output.covhist} basecov={output.basecov} concise \
```
Comment (brwnj) [inline atlas/rules/assemble.snakefile]: I would be more explicit here with concise=t physcov=t

File: atlas/rules/assemble.snakefile [hunk 2]
```diff
@@ -1278,9 +1290,10 @@ rule find_counts_per_region:
         counts = "{sample}/annotation/feature_counts/{sample}_counts.txt"
     params:
         min_read_overlap = config.get("minimum_region_overlap", MINIMUM_REGION_OVERLAP),
-        paired_mode = lambda wc: "-p" if config["samples"][wc.sample].get("paired", True) else "",
-        multi_mapping = "-M" if config.get("count_multi_mapped_reads") else "",
-        primary_only = "--primary" if config.get("primary_only", False) else ""
+        paired_only= "-B" if config.get('cotig_map_paired_only',CONTIG_MAP_PAIRED_ONLY) else "",
```
Comment (brwnj) [inline atlas/rules/assemble.snakefile]: Misspelling of the config key here. Should be "contig_map_paired_only".

**Comment by brwnj** [comment]
Good work on this Silas!

**Pull request merged**